generateData/coreDBQueries.js

```diff
@@ -0,0 +1,62 @@
+const hrtime = require('process.hrtime');
+const pgdb = require('./pgdbConnection.js');
+const csdb = require('./csdbConnection.js');
+
+pgdb.connect();
+// pick a database to test (between 'postgres' and 'cassandra')
+const selectedDataBase = 'cassandra';
+const noOfQueries = 1;
+const reviewId = 33;
+
+async function csExecuteOneAtATime(info, timer) {
+  const query = 'SELECT * FROM reviews WHERE id=?;';
+  const options = { prepare: true };
+
+  while (info.counter++ < info.totalLength) {
+    const params = [reviewId];
+    const result = await csdb.execute(query, params, options);
+    if (result.rows[0]) {
+      await console.log(`${selectedDataBase} query took ${hrtime(timer, 's')}seconds`);
+      // return result.rows[0];
+    }
+  }
+}
+
+const getReviews = async (db) => {
+  const csdbPromises = new Array(noOfQueries);
+  const info = {
+    totalLength: noOfQueries,
+    counter: 0,
+  };
+
+  for (let i = 0; i < noOfQueries; i += 1) {
+    const timer = hrtime();
+
+    if (db === 'postgres') {
+      pgdb.query('SELECT * FROM reviews WHERE id=($1)', [reviewId], (err, result) => {
+        if (err) {
+          console.log(err);
+        } else {
+          console.log(`${selectedDataBase} query took ${hrtime(timer, 's')}seconds`);
+          // return result.rows[0];
+        }
+      });
+    }
+
+    if (db === 'cassandra') {
+      csdbPromises[i] = csExecuteOneAtATime(info, timer);
+      try {
+        // The n promises are going to be resolved when all the executions are completed.
+        await Promise.all(csdbPromises);
+      } finally {
+        await csdb.shutdown();
+      }
+    }
+  }
+};
```

Comment:
Not sure why the loop structure differs from PostgreSQL to Cassandra. Perhaps a comment "just trying two different styles of loops" or "Cassandra works differently than PostgreSQL in that _____ so the loops are different."

Reply:
I spent the whole day trying to figure out some sort of streaming query that works for both. When I originally pushed these code I thought they were good for that until I ran into  some errors afters after pushing. So I was just plugging in things trying to get something going. No luck on both ends